lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,430 @@
+    vault_uri:
+        description:
+            - the URI of the vault for performing operations on keys and secrets.
+    tenant_id:
+        description:
+            - the Azure Active Directory tenant ID that should be used for authenticating requests to the key vault.
+    object_id:
+        description:
+            - the object ID of a user, service principal or security group in the Azure Active Directory tenant for the vault.
+              the object ID must be unique for the list of access policies.
+    application_id:
+        description:
+            - application ID of the client making request on behalf of a principal.
+    permissions:
+        description:
+            - permission templates for access policy.
+        default: Key, Secret, & Certificate Management
+        choices:
+            - Key, Secret, & Certificate Management
+            - Key & Secret Management
+            - Secret & Certificate Management
+            - Key Management
+            - Secret Management
+            - Certificate Management
+            - SQL Server Connector
+            - Azure Backup
+            - Azure Data Lake Store
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Obezimnaka Boms (@ozboms)"
+    - "Ian Philpot (@tripdubroot)"
+'''
+
+EXAMPLES = '''
+##put in examples here
+- name: Create a key vault
+  azure_rm_keyvault:
+    resource_group: Testing
+    name: example.com
+    state: present
+    enabled_for_deployment: True
+    location: westus
+    sku: premium
+
+- name: Create a key secret and certificate management key vault
+  azure_rm_keyvault:
+    name: "ozikeyvault"
+    resource_group: "presentation_rg"
+    state: "present"
+    enabled_for_deployment: True
+    location: "westus2"
+    sku: "premium"
+    enabled_for_disk_encryption: True
+    tenant_id: "72f988bf-86f1-41af-91ab-2d7cd011db47"
+    object_id: "0965570e-4e82-45ae-955a-c655dba6c014"
+
+- name: Create a key management key vault
+  azure_rm_keyvault:
+    name: "ozikeymanagement"
+    resource_group: "presentation_rg"
+    state: "present"
+    enabled_for_deployment: True
+    location: "westus2"
+    sku: "premium"
+    enabled_for_disk_encryption: True
+    tenant_id: "72f988bf-86f1-41af-91ab-2d7cd011db47"
+    object_id: "0965570e-4e82-45ae-955a-c655dba6c014"
+    permissions: "Key Management"
+
+- name: Create a certificate management key vault
+  azure_rm_keyvault:
+    name: "ozicertmanagement"
+    resource_group: "presentation_rg"
+    state: "present"
+    enabled_for_deployment: True
+    location: "westus2"
+    sku: "premium"
+    enabled_for_disk_encryption: True
+    tenant_id: "72f988bf-86f1-41af-91ab-2d7cd011db47"
+    object_id: "0965570e-4e82-45ae-955a-c655dba6c014"
+    permissions: "Certificate Management"
+
+- name: Delete a key vault
+  azure_rm_keyvault:
+    resource_group: Testing
+    name: example.com
+    state: absent
+    location: westus
+
+'''
+
+RETURN = '''
+state:
```

Comment:
We are trying to move away from having the `state` key containing the raw Python SDK return values to just what is not known by the client. In this case I would just return the `id` of the object under the root level, e.g.

```
RETURN = '''
id:
  description:
  returned: success and keyvault exists
  type: str
  example: /subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/presentation_rg/providers/Microsoft.KeyVault/vaults/ozidatamanagement
'''
```

If the user wants to get more information about a resource that is what the facts module is for.